corral/cmd/cmd_fetch.pony

```diff
@@ -4,59 +4,83 @@ use "../bundle"
 use "../util"
 use "../vcs"
 
-class CmdFetch
-  let ctx: Context
-
-  new create(ctx': Context, cmd: Command) =>
-    ctx = ctx'
+primitive CmdFetch
+  fun apply(ctx: Context, cmd: Command) =>
     //ctx.log.info("fetch: " + cmd.string())
     ctx.env.out.print("\nfetch:")
-    match BundleFile.load_bundle(ctx.env, ctx.directory, ctx.log)
-    | let bundle: Bundle =>
-      fetch_bundle_deps(bundle, bundle)
+    match BundleFile.load_bundle(ctx.bundle_dir, ctx.log)
+    | let bundle: Bundle val =>
+      _Fetcher(ctx, bundle).fetch_bundle_deps(bundle)
     | let err: Error =>
       ctx.env.out.print(err.message)
       ctx.env.exitcode(1)
     end
 
-  fun fetch_bundle_deps(base_bundle: Bundle box, bundle: Bundle) =>
-    ctx.log.info("Fetching bundle: " + bundle.name())
+actor _Fetcher
+  let ctx: Context
+  let base_bundle: Bundle val
+
+  new create(ctx': Context, base_bundle': Bundle val) =>
+    ctx = ctx'
+    base_bundle = base_bundle'
+
+  be fetch_bundle_deps(bundle: Bundle val) =>
+    ctx.log.info("Fetching direct deps of bundle: " + bundle.name())
     for dep in bundle.deps.values() do
-      try
+      //try
```

Comment:
what's up with the commented out code here?

should it be deleted?

Reply:
Oops, yes. That was the old way.